src/main/java/com/github/manolo8/darkbot/core/manager/PetManager.java

```diff
@@ -383,7 +385,10 @@ private void updateNpcLocatorList(long gearsSprite) {
         if (locatorNpcList.getSize() < oldSize && validUntil > System.currentTimeMillis()) return;
 
         validUntil = System.currentTimeMillis() + 100;
+        int oldState = locatorList.hashCode();
         locatorNpcList.sync(locatorList, Gear::new, null);
+        if (oldState != locatorList.hashCode())
```

Comment:
Technically speaking hashCode and equals work differently.
Even if hash code is not different, they could be different lists, because hash collisions exist.

Even if it's not nearly as efficient, i think here you should either:
A) create a copy of locator list before the clone, and compare with `equals` to the list after the sync
B) modify sync to return if anything has changed (this may be complicated as sync is already used in other situations where the result needs to be the returned list